webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioPlaylistDialog/index.tsx

```diff
@@ -0,0 +1,242 @@
+import * as RA from "ramda-adjunct";
+import { useMemo, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useUpdateEffect } from "react-use";
+import usePromise from "../../../../../../../../hooks/usePromise";
+import type { StudyMetadata } from "../../../../../../../../types/types";
+import BasicDialog from "../../../../../../../common/dialogs/BasicDialog";
+import type { SubmitHandlerPlus } from "../../../../../../../common/Form/types";
+import UsePromiseCond from "../../../../../../../common/utils/UsePromiseCond";
+import ScenariosSelectionFE from "./YearsSelectionFE";
+interface Props {
+  study: StudyMetadata;
+  open: boolean;
+  onClose: VoidFunction;
+}
+
+function ScenarioPlaylistDialog({ study, open, onClose }: Props) {
+  const { t } = useTranslation();
+  const dataGridApiRef = useRef<DataGridFormApi<PlaylistData>>(null);
+  const [yearsSelection, setYearsSelection] = useState<number[] | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isDirty, setIsDirty] = useState(false);
+  const [totals, setTotals] = useState({ selected: 0, sumWeights: 0 });
+  const closeAction = useConfirm();
+  const res = usePromise(() => getPlaylistData({ studyId: study.id }), [study.id]);
+  const disableBtnGroup = isSubmitting || (yearsSelection !== null && yearsSelection.length === 0);
+
+  const columns = useMemo<DataGridFormProps<PlaylistData>["columns"]>(() => {
+    return [
+      {
+        id: "status" as const,
+        title: t("global.status"),
+        grow: 1,
+        trailingRowOptions: {
+          hint: `${translateWithColon(t("study.configuration.general.mcScenarioPlaylist.totalSelected.label"))} ${totals.selected}`,
+        },
+      },
+      {
+        id: "weight" as const,
+        title: t("global.weight"),
+        grow: 1,
+        trailingRowOptions: {
+          hint: `${translateWithColon(t("study.configuration.general.mcScenarioPlaylist.totalWeight.label"))} ${totals.sumWeights}`,
+        },
+      },
+    ];
+  }, [t, totals]);
+
+  useUpdateEffect(() => {
+    if (res.data) {
+      updateTotals(res.data);
+    }
+  }, [res.data]);
+
+  ////////////////////////////////////////////////////////////////
+  // Utils
+  ////////////////////////////////////////////////////////////////
+
+  const mapSelectedYears = (fn: (item: Playlist) => Playlist, data: PlaylistData): PlaylistData => {
+    return RA.mapIndexed((item, index) => {
+      if (yearsSelection === null || yearsSelection.includes(index + 1)) {
+        return fn(item);
+      }
+      return item;
+    }, data);
+  };
+
+  function updateTotals(data: PlaylistData) {
+    setTotals(
+      Object.entries(data).reduce(
+        (acc, [_, { status, weight }]) => {
+          if (status) {
+            acc.selected += 1;
+            acc.sumWeights += weight;
+          }
+          return acc;
+        },
+        { selected: 0, sumWeights: 0 },
+      ),
+    );
+  }
```

Comment:
you can pre-calculate the totals before calling `setTotals()`, which makes the code slightly more readable, and memoize `updateTotals` as it's passed to `DataGridForm`

Reply:
I added a variable, but no need to memoize here, it's only used by `DataGridForm` not `DataGrid`

Reply:
ok